File: my-folder/0254-factor-combinations/solution.java
class Solution {
    public void backtrack(List<List<Integer>> ans,List<Integer> list ,int n, int start)
    {
        if(n<2) {
            if(list.size()>1)
            {
                ans.add(new ArrayList<Integer>(list));
            }
            return;
        }
        for(int i=start;i<=n;i++)
        {
            if(n%i==0)
            {
                list.add(i);
                backtrack(ans,list,n/i,i);
                list.remove(list.size()-1);
            }
        }
    }
    public List<List<Integer>> getFactors(int n) {
        List<List<Integer>> ans=new ArrayList<>();
        backtrack(ans,new ArrayList<>(),n,2);
        return ans;
    }
}

File: my-folder/0254-factor-combinations/solution.swift
class Solution {
    func getFactors(_ n: Int, _ start: Int = 2) -> [[Int]] {
    if n == 2 {
        return []
    }
    var ans = [[Int]]()
    for i in stride(from: start, through: n/2, by: 1) {
        if (n%i) == 0 {
            let next = n/i
            if next >= i {
                var temp = getFactors(next, i) + [[next]]
                temp = temp.map {
                    return ([i] + $0)
                }
                ans += temp
            }
        }
    }
    return ans
}
}
